File: boj/15649.py
import sys
input = sys.stdin.readline

n, m = list(map(int, input().split()))
trace = []


def dfs(k):
    if k == m:
        print(' '.join(list(map(str, trace))))
        return
    for i in range(n):
        if i+1 not in trace:
            trace.append(i+1)
            dfs(k+1)
            trace.pop()


dfs(0)

File: boj/15649.js
function nAndM(n, m, trace) {
  if (trace.length === m) {
    console.log(trace.join(' '))
    return
  }
  for (let i = 1; i < n + 1; i++) {
    if (!trace.includes(i)) {
      trace.push(i)
      nAndM(n, m, trace)
      trace.pop()
    }
  }
}

nAndM(4, 4, [])
